tests/PKSim.Tests/Presentation/RenameSubjectUICommandSpecs.cs

```diff
@@ -0,0 +1,94 @@
+﻿using FakeItEasy;
+using OSPSuite.BDDHelper;
+using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Presentation.Views;
+using PKSim.Core;
+using PKSim.Core.Model;
+using PKSim.Presentation.UICommands;
+
+namespace PKSim.Presentation
+{
+   public abstract class concern_for_RenameSubjectUICommand : ContextSpecification<RenameSubjectUICommand>
+   {
+      protected RenameBuildingBlockUICommand _renameBuildingBlockCommand;
+      protected RenameObjectUICommand _renameObjectCommand;
+      protected RenameObservedDataUICommand _renameObservedDataCommand;
+      protected IMdiChildView _mdiChildView;
+      protected ISingleStartPresenter _mdiChildPresenter;
+
+      protected override void Context()
+      {
+         _renameBuildingBlockCommand= A.Fake<RenameBuildingBlockUICommand>();
+         _renameObjectCommand= A.Fake<RenameObjectUICommand>();
+         _renameObservedDataCommand= A.Fake<RenameObservedDataUICommand>();
+         _mdiChildView= A.Fake<IMdiChildView>();
+         _mdiChildPresenter= A.Fake<ISingleStartPresenter>();
+         A.CallTo(() => _mdiChildView.Presenter).Returns(_mdiChildPresenter);
+         sut = new RenameSubjectUICommand(_renameBuildingBlockCommand, _renameObservedDataCommand, _renameObjectCommand) {Subject = _mdiChildView};
+
+      }
+      protected override void Because()
+      {
+         sut.Execute();
+      }
+
+   }
+   //TODO Uncomment those tests when Execute is made Virtual in OSPSuite
```

Comment:
The execute method of ObjectBaseCommand is not virtual and cannot be intercepted. I'll change it when I update OSPSuite.Core with new smartxls and simmodel